add largeTitleDisplayMode to PaymentContext

bg-stripe opened the pull request:
r? @bdorfman-stripe 

`PaymentContext` will set any pushed or presented VC's `largeTitleDisplayMode` to its own. 

re: #830 – I wasn't able to reproduce the layout issue, but this does fix not being able to set `largeTitleDisplayMode` of pushed PaymentContext view controllers to `never`.

bg-stripe commented on Stripe/STPPaymentContext.m:
```diff
@@ -292,9 +293,11 @@ - (void)presentPaymentMethodsViewControllerWithNewState:(STPPaymentContextState)
 
             UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:paymentMethodsViewController];
             navigationController.navigationBar.stp_theme = self.theme;
+            navigationController.navigationBar.prefersLargeTitles = YES;
```
Based on the docs, it looks like it isn't necessary to expose this property on `PaymentContext` too – just setting it to always `YES` always seems fine.

> When this property is set to true, the navigation bar allows the title to be displayed out-of-line and using a larger font. The navigation item used to build the bar must specify whether it wants its title displayed in the large or small format. Use the largeTitleDisplayMode property to configure the title's appearance.

> When the property is set to false, the navigation bar displays the title inline with the other bar button items.

bg-stripe commented on Example/Standard Integration (Swift)/BrowseProductsViewController.swift:
```diff
@@ -35,6 +35,7 @@ class BrowseProductsViewController: UITableViewController {
         super.viewDidLoad()
         self.navigationItem.title = "Emoji Apparel"
         self.navigationController?.navigationBar.isTranslucent = false
+        self.navigationController?.view.backgroundColor = .white
```
This fixes the transition when `PaymentContext` pushes a small title view controller onto a large title nav controller (there's a black flash without this line).

bdorfman-stripe commented on Stripe/PublicHeaders/STPPaymentContext.h:
```diff
@@ -233,6 +233,14 @@ NS_ASSUME_NONNULL_BEGIN
+ which causes the title to use the same styling as the previously displayed
+ navigation item.
+ */
+@property (nonatomic, assign) UINavigationItemLargeTitleDisplayMode largeTitleDisplayMode;
```
Might want to consider adding in some extra notes about how this property interacts with push vs present. Eg. the default of `automatic` won't work for present because there is no previous item. And for push we don't alter the owning navigation bar, you have to make sure it's navigation bar has `prefersLargeTitles = YES` in order for this to do anything.

bg-stripe replied:
ah, good call – I'll expand

bg-stripe commented:
> Do we need to update our tests to cover this? Either regular unit tests or snapshot tests? (I made a bunch of snapshot test updates in the xcode9 branch if you want to wait for that to be merged).

Ah yeah, now's probably a good time to add some PaymentContext snapshot tests. I'll wait and update the PR after #851 gets in

bg-stripe commented:
I've updated docs, added availability guards, and added some snapshot tests (b10520a) for PaymentContext pushing payment methods / shipping onto a hostViewController.

re-r? @bdorfman-stripe

bdorfman-stripe commented on Tests/Tests/FBSnapshotTestCase+STPViewControllerLoading.h:
```diff
@@ -9,5 +9,18 @@
 #import <FBSnapshotTestCase/FBSnapshotTestCase.h>
 
 @interface FBSnapshotTestCase (STPViewControllerLoading)
-- (UIView *)stp_preparedAndSizedViewForSnapshotTestFromViewController:(UIViewController *)viewController;
```
Might be nice to retain this, or a similarly named method, that encompasses both the other helpers (since everything except Pay Context immediately calls both in a row I think?)

bg-stripe replied:
ah yeah, that's a good call, I'll update
